rollout/sync.go

```diff
@@ -102,6 +104,14 @@ func (c *rolloutContext) syncReplicaSetRevision() (*appsv1.ReplicaSet, error) {
 		condition := conditions.NewRolloutCondition(v1alpha1.RolloutProgressing, corev1.ConditionTrue, conditions.FoundNewRSReason, msg)
 		conditions.SetRolloutCondition(&c.rollout.Status, *condition)
 		updatedRollout, err := c.argoprojclientset.ArgoprojV1alpha1().Rollouts(c.rollout.Namespace).UpdateStatus(ctx, c.rollout, metav1.UpdateOptions{})
+		// adding this block as argoprojclientset update deletes typemeta fields.
```

Comment:
This is because of generated code:
https://github.com/argoproj/argo-rollouts/blob/e42b4d4b9d5a5b8f2334745aeb9b9692d09bf60c/pkg/client/clientset/versioned/typed/rollouts/v1alpha1/rollout.go#L127

The result here doesnot have typemeta fields and that is being propogated to rollout when returned from this update call.

Reply:
As this code gets generated from k8s.io/code-generator I think it is a bug there.